src/DataTomeAnalysis.h

```diff
@@ -275,6 +275,56 @@ class DataTomeAnalysis : public DataTomeMvAvg<TypeOfArray, TypeOfSum> {
     return sqrt(partial_var(partial_id) /
                 this->partial_point_count(partial_id));
   }
+
+  private:
+    TypeOfArray medianOfMedians(int l, int r, TypeOfArray nums[]) {
+      int k = 5;
+      size_t size = r - l + 1,
+             medians_size = (size + k - 1) / k;
+      TypeOfArray medians[medians_size];
+      for (int i = l, m_count = 0; i <= r; i += k, m_count++)
+      {
+        int left = i, right = min(i + k, r + 1),
```

Comment:
It needs to define a min() function. The compiler also recommended using `std::min()`, it may work, but I don't know if it works on all devices, so running the building tests is advised.